.spotbugs/spotbugs-exclude.xml

```diff
@@ -1,5 +1,20 @@
 <?xml version="1.0" encoding="UTF-8"?>
 <FindBugsFilter>
+  <Match>
+    <Bug pattern="DMI_HARDCODED_ABSOLUTE_FILENAME"/>
+  </Match>
+  <Match>
+    <Bug pattern="DM_EXIT"/>
+    <Package name="io.strimzi.operator.cluster"/>
+  </Match>
+  <Match>
+    <Bug pattern="DM_EXIT"/>
+    <Package name="io.strimzi.operator.user"/>
+  </Match>
+  <Match>
+    <Bug pattern="BX_UNBOXING_IMMEDIATELY_REBOXED"/>
+    <Package name="io.strimzi.operator.cluster.operator.assembly"/>
+  </Match>
```

Comment:
Why is it fine to exclude all these warnings?

Reply:
It should be excluded just in the `io.strimzi.operator.cluster.operator.assembly` package. It is because of this line `ss -> ss == null ? 0 : ss.getSpec().getReplicas()` in `io.strimzi.operator.cluster.operator.assembly.KafkaAssemblyOperator.ReconciliationState#zkScaleUpStep`

Reply:
So what do you think that means and how are you going to fix it?

(I have a pretty good idea, but I'm feeling mean).

Reply:
Oh, and what about the rest of these disabled checks ^^?

Reply:
I tried but I did not find any solution. Be mean and suggest me any idea, please.

Reply:
The `0` branch of the ternary operator is unboxed, but `StatefulSetSpec.getReplicas()` returns an `Integer`. The types of each branch must agree, so I suspect that javac is unboxing the result of `getReplicas()` so that the result of the ternary operation is a primitive. But (I'm guessing) the result of the ternary is being used in a lambda whose method type has a reference type as its result. So the primitive is immediately boxed again. I suspect you can avoid the warning by forcing the result type of the ternary to be a reference type. Do that by forcing the primitive branch to be boxed, i.e. change the `0` to `Integer.valueOf(0)`.

Reply:
Thank you.